### ISSUE TITLE
What if properties of a class are also generated by another code generator?

### ISSUE THREAD

**Issue by myd7349** [diegofrata/Generator.Equals#35]
Hi! Thank you for this awesome library.

Suppose I have a ViewModel class like this:

```csharp
using System.IO;
using System.Windows.Media.Imaging;

using CommunityToolkit.Mvvm.ComponentModel;
using Dapper.Contrib.Extensions;
using Generator.Equals;

namespace CoilQueryTool.ViewModels
{
    [Equatable(IgnoreInheritedMembers = true)]
    [Table("CoilMap")]
    internal partial class CoilRecordViewModel : ObservableObject
    {
        public CoilRecordViewModel ShallowCopy()
        {
            return (CoilRecordViewModel)MemberwiseClone();
        }

        [IgnoreEquality]
        public int Id { get; set; }

        [ObservableProperty]
        private string coilId = null!;

        [ObservableProperty]
        private string? name;

        [ObservableProperty]
        private string? connector;

        [ObservableProperty]
        private string? pN;

        [ObservableProperty]
        private string? description;

        [ObservableProperty]
        private string? machine;

        [ObservableProperty]
        private string? mode;
    }
}
```

Since most of the properties of this class is generated by another code generator - CommunityToolkit.Mvvm's code generator, Generator.Equals will not generate code for these `properties`.

I am wondering if `[DefaultEquality]` can be used like this:

```csharp
class MyViewModel
{
    [DefaultEquality]
    [ObservableProperty]
    private string title;
}
```

**Comment by diegofrata** [comment]
This use case is not currently supported. Source generators cannot interact in this manner. 

I think the only option here would be to extend Generator.Equals to support fields in addition to properties, but I am not sure how much work would that involve.

**Comment by diegofrata** [comment]
I had a look at the code, I don't think it would be too hard to add support for fields. I can give it a go at some point this week.

**Comment by myd7349** [comment]
Hi! @diegofrata Thank you for your quick reply.

**Comment by diegofrata** [comment]
Sorry, did not have time last week. Will try this one.

**Comment by diegofrata** [comment]
This is now available in 2.7.0.

**Comment by myd7349** [comment]
Hi! @diegofrata It is amazing! Thank you very much.
